### PR TITLE
[PLGN-429] - Sentinelone -Adding new action for move agents between sites

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - To add a new actions that will allow for agents to be moved between sites
  - To add in new unit tests to cover this action

https://issues.corp.rapid7.com/browse/PLGN-429

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

```
❯ python -m unittest *
.
.
Ran 175 tests in 2.072s
```

all unit tests are passing for the plugin 

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

testing local in docker with the following input we can see the agent moved site as expected

"input": {
    "targetSiteId" : "162************",            
    "filter": {"computerName": "WindowsX64"}
} 

before 

image

after 

image

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/sentinelone/plugin.spec.yaml [hunk 1]
```diff
@@ -3044,6 +3044,29 @@ actions:
         type: integer
         required: true
         example: 1
+  move_between_sites:
+    title: Move Agent to Another Site
+    description: Move an agent to another Site
+    input:
+      targetSiteId:
+        title: Target Site ID
+        description: A id of the new site to move the agents to
```
Comment (igorski-r7) [inline plugins/sentinelone/plugin.spec.yaml:3053]:
```suggestion
        description: A ID of the new site to move the agents to
```

File: plugins/sentinelone/help.md [hunk 2]
```diff
@@ -2097,6 +2135,7 @@ Example output:
 
 # Version History
 
+* 9.1.0 - Add new actions `Move Agent to Another Site`
```
Comment (cmcnally-r7) [inline plugins/sentinelone/help.md:2138]:
```suggestion
* 9.1.0 - `Move Agent to Another Site`: Action added
```

File: plugins/sentinelone/plugin.spec.yaml (hunk 1 above)
Comment (cmcnally-r7) [reply plugins/sentinelone/plugin.spec.yaml:3053]: Just to follow up on this one
```suggestion
        description: The ID of the new site to move the agents to
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]
Other than that lgtm

File: plugins/sentinelone/plugin.spec.yaml (hunk 1 above)
Comment (rbowden-r7) [reply plugins/sentinelone/plugin.spec.yaml]: This has been updated

File: plugins/sentinelone/help.md (hunk 2 above)
Comment (rbowden-r7) [reply plugins/sentinelone/help.md]: This has been reworded

**Review by cmcnally-r7 — APPROVED** [review]

**Review by dlaverty-r7 — COMMENTED** [review]
@rbowden-r7 changes look good but are there any additional required permissions for this action? If so can you please add in some info in the description or troubleshooting guide?

**Comment by rbowden-r7** [comment]
> @rbowden-r7 changes look good but are there any additional required permissions for this action? If so can you please add in some info in the description or troubleshooting guide?

from there docs it syas `This command requires Account or Global level access.` so i'll add that in

File: plugins/sentinelone/plugin.spec.yaml [hunk 3]
```diff
@@ -3044,6 +3044,29 @@ actions:
         type: integer
         required: true
         example: 1
+  move_between_sites:
+    title: Move Agent to Another Site
+    description: Move an agent to another Site, This command requires Account or Global level access for your user role
```
Comment (dlaverty-r7) [inline plugins/sentinelone/plugin.spec.yaml:3049]: Suggested reword: This command -> This action

Comment (rbowden-r7) [reply plugins/sentinelone/plugin.spec.yaml]: This has been updated

**Review by dlaverty-r7 — APPROVED** [review]

**Pull request merged**